clang/test/Modules/autolink_private_module.m

```diff
@@ -0,0 +1,25 @@
+// Test that autolink hints for frameworks don't use the private module name.
```

Comment:
Why is this desired? What happens if the private module name is provided?

Reply:
The auto-link hint is supposed to provide a framework name, which is incorrect when you pass the name of the private module. The linker ignores invalid auto-link hints, but it might emit warnings about them when the link fails for other reasons. Besides the warning messages being confusing, this could also cause link errors if the public module was never used (very rare) and therefore the framework was never linked.

Reply:
Makes sense, thanks!